plugins/woocommerce/templates/checkout/form-login.php

```diff
@@ -12,25 +12,44 @@
  *
  * @see https://woocommerce.com/document/template-structure/
  * @package WooCommerce\Templates
- * @version 3.8.0
+ * @version 9.9.0
  */
 
 defined( 'ABSPATH' ) || exit;
 
-if ( is_user_logged_in() || 'no' === get_option( 'woocommerce_enable_checkout_login_reminder' ) ) {
+$registration_at_checkout   = WC_Checkout::instance()->is_registration_enabled();
+$login_reminder_at_checkout = 'yes' === get_option( 'woocommerce_enable_checkout_login_reminder' );
+
+if ( is_user_logged_in() ) {
 	return;
 }
 
-?>
-<div class="woocommerce-form-login-toggle">
-	<?php wc_print_notice( apply_filters( 'woocommerce_checkout_login_message', esc_html__( 'Returning customer?', 'woocommerce' ) ) . ' <a href="#" class="showlogin">' . esc_html__( 'Click here to login', 'woocommerce' ) . '</a>', 'notice' ); ?>
-</div>
-<?php
+if ( $login_reminder_at_checkout ) : ?>
+	<div class="woocommerce-form-login-toggle">
+		<?php
+		wc_print_notice(
+			apply_filters( 'woocommerce_checkout_login_message', esc_html__( 'Returning customer?', 'woocommerce' ) ) . // phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingHookComment
+			' <a href="#" class="showlogin">' . esc_html__( 'Click here to login', 'woocommerce' ) . '</a>',
+			'notice'
+		);
+		?>
+	</div>
+	<?php
+endif;
+
+if ( $registration_at_checkout || $login_reminder_at_checkout ) :
+
+	$hide_form = true;
+	// Always show the form after a login attempt.
+	if ( isset( $_SERVER['REQUEST_METHOD'] ) && 'POST' === $_SERVER['REQUEST_METHOD'] && isset( $_POST['login'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$hide_form = false;
+	}
```

Comment:
I think you can simplify this quite a bit i.e. 

```suggestion
	// Always show the form after a login attempt.
	$hide_form = ! isset( $_POST['login'] ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
```

I'm not sure the other logic adds much value here. Might be more readable if you flip the logic too (_show_form_).